src/api/spec/helpers/webui/webui_helper_spec.rb

```diff
@@ -319,7 +319,69 @@
   end
 
   describe '#user_and_role' do
-    skip('Please add some tests')
+    let(:user) { create(:user) }
+    let(:nobody) { create(:user_nobody) }
+
+    before do
+      User.current = user
+    end
+
+    context 'for users' do
+      it 'display the users realname with its login' do
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user.realname}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.realname} (#{user.login})</a>"
+        )
+      end
+
+      it 'falls back to users login if realname is empty' do
+        user.realname = ''
+        user.save!
+
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.login}</a>"
+        )
+      end
+    end
+
+    context 'with options' do
+      it 'does not show an icon if option disables it' do
+        expect(user_and_role(user.login, nil, no_icon: true)).to eq(
+          "<a href=\"/user/show/#{user.login}\">#{user.realname} (#{user.login})</a>"
+        )
+      end
+
+      it 'only shows user login if short option is set' do
+        user.realname = ''
+        user.save!
+
+        expect(user_and_role(user.login, nil, short: true)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.login}</a>"
+        )
+      end
+    end
+
+    context 'for guests' do
+      it 'does not link to user profiles if current user is nobody' do
+        User.current = nobody
+
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user.realname}\" " + \
+          "src=\"/user/icon/#{user.login}?size=20\" />#{user.realname} (#{user.login})"
+        )
+      end
+    end
+
+    context 'with roles' do
```

Comment:
I see!
So the only difference between a logged in user and a nobody user is displaying the link, right?
So I would move everything into a shared_example and then test the shared example for both cases. For the logged in case I would test additional that there is a link.

Also it might sense to use regular expression matcher then.
https://relishapp.com/rspec/rspec-expectations/v/2-2/docs/matchers/match-matcher